torchdrivesim/simulator.py

```diff
@@ -1287,6 +1313,9 @@ def make_full_action(replay_action, mask, given_action):
         )
         self.update_present_mask(non_replay_present_mask)
 
+        innermost_simulator = self.get_innermost_simulator()
+        innermost_simulator.traffic_controls_step()
```

Comment:
This is a little confusing to me. We're already stepping the inner simulator, so wouldn't this step the traffic controls twice?

Reply:
Good catch. Indeed it's a duplicate. Luckily, it didn't affect the experiments, because it called `step(self, time: int)` inside the traffic control twice with the same `time`. Two calculation will return the same result. Let me delete these two lines.